### PR TITLE
Fix up intro doc

### PR DESCRIPTION
**Pull request by Manishearth**
I've made the existence of testdata more prominent so that it's clearer that there's a way to play around with ICU4X without generating your own data. Otherwise I've updated the example so that it still works.

Fixes https://github.com/unicode-org/icu4x/issues/1719

### CODE REVIEW

File: docs/tutorials/intro.md [hunk 1]
```diff
@@ -182,7 +154,7 @@ and then we can use it in our code:
 use icu_provider_fs::FsDataProvider;
 
 fn main() {
-    let _provider = FsDataProvider::try_new("/home/{USER}/projects/icu/icu4x-data")
+    let _provider = FsDataProvider::try_new("/path/to/icu4x/provider/testdata/data/json")
```
Comment (sffc) [inline docs/tutorials/intro.md:157]: Issue: They should use the APIs exported from the icu_testdata crate, like `icu_testdata::get_provider()`.

(Note that this is the API that embeds the absolute path to the data file)

**Review by sffc — CHANGES_REQUESTED** [review]

File: docs/tutorials/intro.md (hunk 1 above)
Comment (Manishearth) [reply docs/tutorials/intro.md:157]: I think the point of this example is to demonstrate data loading in general; whether it be from test data or from personally generated data. To me, the testdata here is just one example of a data source that's convenient to use.

Comment (sffc) [reply docs/tutorials/intro.md:157]: Ok. However, I think the risk of people thinking that this is the right way to use testdata, and putting these paths into their test file, is greater than the benefit of teaching them about this API. We could link to the FsDataProvider docs which has these examples.

Comment (Manishearth) [reply docs/tutorials/intro.md:157]: I feel like there _is_ no "right way" to use testdata; the crate exists for our internal test uses, this is just for exploratory purposes. I can tighten up the language about this.

Comment (sffc) [reply docs/tutorials/intro.md:157]: > I feel like there is no "right way" to use testdata

I disagree. `icu_testdata` is a perfectly fine crate to use if you understand its limitations, and it has public APIs to help using it.

https://docs.rs/icu_testdata/0.2.0/icu_testdata/

**Review by sffc — APPROVED** [review]

File: docs/tutorials/intro.md [hunk 2]
```diff
@@ -223,12 +183,74 @@ fn main() {
...
+wget https://github.com/unicode-org/icu/releases/download/release-70-1/icuexportdata_uprops_full.zip
+unzip icuexportdata_uprops_full.zip
+git clone https://github.com/unicode-org/icu4x
+cd icu4x
+git checkout icu@0.5.0
+cargo run --bin icu4x-datagen -- \
+    --cldr-tag 40.0.0 \
+    --uprops-root ../icuexportdata_uprops_full/small \
+    --out ~/projects/icu/icu4x-data \
+    --all-keys --all-locales
```
Comment (robertbastian) [inline docs/tutorials/intro.md:235]: The uprops zip doesn't have to be manually downloaded anymore. 

Also, `--all-keys` doesn't work with ICU70 as the script extensions data is not included, so we should point this to either release-71-rc or release-71-1. 

```suggestion
git clone https://github.com/unicode-org/icu4x
cd icu4x
git checkout icu@0.5.0
cargo run --bin icu4x-datagen --features download -- \
    --cldr-tag 40.0.0 \
    --uprops-tag release-71-1 \
    --out ~/projects/icu/icu4x-data \
    --all-keys --all-locales
```

File: docs/tutorials/intro.md [hunk 3]
```diff
@@ -223,12 +183,74 @@ fn main() {
...
+* Then we pass `--cldr-tag` which informs the program which CLDR version to use
+* Then we pass `--uprops-root` which informs the program where to get the Unicode property data
+* Then we pass `--out` directory which is where the generated ICU4X data will be stored
+* Finally, we set `--all-keys` which specify that we want to export all keys available
+
+After that step, it should be possible to navigate to `~/projects/icu/icu4x-data` and there should be a `manifest.json` file, and directories with data.
+
+*Notice:* In this tutorial we export data as compact `JSON` which provides decent performance and readable data files. There are other formats and options for formatting of the data available. Please consult `cargo run --bin icu4x-datagen -- --help` for details.
+*Notice:* In particular, in production, the `postcard` format will yield better performance results.
+*Notice:* For offline or unconventional use, the user can also pass `--cldr-core` and `--cldr-dates` paths to local clones of the repositories instead of `--cldr-tag`.
```
Comment (robertbastian) [inline docs/tutorials/intro.md:252]: hm? Do you mean the user can pass a custom CLDR directory with `--cldr-root`?

Comment (Manishearth) [reply docs/tutorials/intro.md]: Yeah that's legacy stuff that was already present, fixed.

File: docs/tutorials/intro.md (hunk 2 above)
Comment (robertbastian) [reply docs/tutorials/intro.md]: Also fix up the explanation of the command below.

File: docs/tutorials/intro.md (hunk 3 above)
Comment (robertbastian) [reply docs/tutorials/intro.md]: If we're calling `--cldr-root` "unconventional", maybe we should make the `"download"` feature default? @sffc

File: docs/tutorials/intro.md (hunk 2 above)
Comment (robertbastian) [reply docs/tutorials/intro.md]: And this also removes the `cd ~/projects/icu`, so don't tell the user to navigate there.

File: docs/tutorials/intro.md [hunk 4]
```diff
@@ -223,12 +183,71 @@ fn main() {
 If all went well, running the app with `cargo run` should display:
 
 ```
-📅: 14 października 2020 13:21:00
+📅: 2020年10月14日 13:21:00
```
Comment (robertbastian) [inline docs/tutorials/intro.md:186]: update line 256

File: docs/tutorials/intro.md (hunk 2 above)
Comment (Manishearth) [reply docs/tutorials/intro.md]: There's still a cd up above

File: docs/tutorials/intro.md [hunk 5]
```diff
@@ -238,8 +238,7 @@ The last command is a bit dense, so let's dissect it.
 * We tell it that the binary is named `icu4x-datagen`
 * Then we use `--` to separate arguments to `cargo` from arguments to our app
 * Then we pass `--cldr-tag` which informs the program which CLDR version to use
-* Then we pass `--uprops-root` which informs the program where to get the Unicode property data
-* Then we pass `--out` directory which is where the generated ICU4X data will be stored
+* Then we pass `--out` directory which is where we want the generated ICU4X data to be stored
```
Comment (robertbastian) [inline docs/tutorials/intro.md:241]: ... and a `--uprops-tag` which informs the program which ICU version to use

**Review by robertbastian — APPROVED** [review]

**Review by robertbastian — APPROVED** [review]

**Review by sffc — APPROVED** [review]
Still fine for me

**Pull request merged**